packages/react-native/Libraries/AppDelegate/RCTAppDelegate.mm

```diff
@@ -144,20 +144,18 @@ - (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
   return nil;
 }
 
-- (NSDictionary *)extraInitialProps
+- (void)updateInitialProps:(NSMutableDictionary *)initialProps
 {
-  NSMutableDictionary *initProps = self.initialProps ? [self.initialProps mutableCopy] : [NSMutableDictionary new];
 #ifdef RCT_NEW_ARCH_ENABLED
   // Hardcoding the Concurrent Root as it it not recommended to
   // have the concurrentRoot turned off when Fabric is enabled.
-  initProps[kRNConcurrentRoot] = @([self fabricEnabled]);
+  initialProps[kRNConcurrentRoot] = @([self fabricEnabled]);
 #endif
-  return initProps;
 }
```

Comment:
Best practice for private methods that need no instance state is to just make them static

```
static NSDictionary *updateInitialProps(NSDictionary *initialProps, BOOL isFabricEnabled) {
#ifdef RCT_NEW_ARCH_ENABLED
  NSMutableDictionary *mutableProps = [initialProps mutableCopy] ?: [NSMutableDictionary new];
  // Hardcoding the Concurrent Root as it it not recommended to
  // have the concurrentRoot turned off when Fabric is enabled.
  mutableProps[kRNConcurrentRoot] = @(isFabricEnabled);
  return mutableProps;
#else
  return initialProps;
#endif
}
```